pandas/tseries/period.py

```diff
@@ -305,7 +306,7 @@ def _simple_new(cls, values, name=None, freq=None, **kwargs):
             if (len(values) > 0 and is_float_dtype(values)):
                 raise TypeError("PeriodIndex can't take floats")
             else:
-                return PeriodIndex(values, name=name, freq=freq, **kwargs)
+                return cls(values, name=name, freq=freq, **kwargs)
```

Comment:
It's not what's causing the test failure (it could only be different if `cls != PeriodIndex`